L1Trigger/TrackFindingTMTT/src/TrkRZfilter.cc

```diff
@@ -0,0 +1,368 @@
+        }
+      }
+
+      //--- Filter stubs assigned to track, checking they are consistent with requested criteria.
+
+      // Get debug printout for specific regions.
+      bool print = false;
+      // unsigned int mbin = trkIN.getCellLocationHT().first;
+      // unsigned int cbin = trkIN.getCellLocationHT().second;
+      // if(mbin == 0 && cbin == 45 && trkIN.iEtaReg() == 14 && trkIN.iPhiSec() == 6) print = true;
+      // cout << "track in region "<<trkIN.iEtaReg()<<", "<<trkIN.iPhiSec()<< " bin " << mbin << ", "<< cbin << endl;
+
+      vector<const Stub*> filteredStubs = stubs;
+      if (rzFilterName_ == "SeedFilter") {
+        filteredStubs = this->seedFilter(filteredStubs, trkIN.qOverPt(), print);
+      } else {
+        throw cms::Exception("TrkRzFilter: ERROR unknown r-z track filter requested") << rzFilterName_ << endl;
+      }
+
+      // Check if track still has stubs in enough layers after filter.
+
+      unsigned int numLayersAfterFilters = Utility::countLayers(settings_, filteredStubs);
+      if (numLayersAfterFilters < settings_->minFilterLayers())
+        trackAccepted = false;
+      //if (numLayersAfterFilters < Utility::numLayerCut("SEED", settings_, iPhiSec_, iEtaReg_, fabs(trkIN.qOverPt()))) trackAccepted = false;
+
+      if (trackAccepted) {
+        // Estimate r-z helix parameters from centre of eta-sector if no estimate provided by r-z filter.
+        if (!rzHelix_set_)
+          this->estRZhelix();
+
+        pair<float, float> helixRZ(rzHelix_z0_, rzHelix_tanL_);
+
+        // Create copy of original track, except now using its filtered stubs, to be added to filteredTrack collection.
+        L1track3D trkOUT(settings_,
+                         filteredStubs,
+                         trkIN.getCellLocationHT(),
+                         trkIN.getHelix2D(),
+                         helixRZ,
+                         trkIN.iPhiSec(),
+                         trkIN.iEtaReg(),
+                         trkIN.optoLinkID(),
+                         trkIN.mergedHTcell());
+
+        filteredTracks.push_back(trkOUT);
+      }
+    }
+
+    // Optionally run duplicate track removal on all the 3D tracks found in this sector & store final 3D track collection.
+    filteredTracks = killDupTrks_.filter(filteredTracks);
+
+    return filteredTracks;
+  }
+
+  //=== Use Seed Filter to produce a filtered collection of stubs on this track candidate that are consistent with a straight line
+  //=== in r-z using tracklet algo.
+
+  vector<const Stub*> TrkRZfilter::seedFilter(const std::vector<const Stub*>& stubs, float trkQoverPt, bool print) {
+    unsigned int numLayers;                      //Num of Layers in the cell after that filter has been applied
+    std::vector<const Stub*> filtStubs = stubs;  // Copy stubs vector in filtStubs
+    bool FirstSeed = true;
+    const int FirstSeedLayers[] = {1, 2, 11, 21, 3, 12, 22, 4};           //Allowed layers for the first seeding stubs
```

Comment:
No, it can't be obtained from the geometry. It defines which layers are used to find a pair of stubs to seed tracks. They were chosen after optimisation. 

Replaced container with "static const vector<unsigned int>", which has the advantage over array that it automatically adjusts its size to match the elements in the initialization list.